connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -276,17 +422,33 @@ async function deleteIssue(
     throw new Error("Connector not found");
   }
 
-  localLogger.info("Deleting GitHub issue from Dust Data Source.");
-  await deleteFromDataSource(
-    dataSourceConfig,
-    getIssueDocumentId(repoId, issueNumber)
+  const issueInDb = await GithubIssue.findOne({
+    where: {
+      repoId: repoId.toString(),
+      issueNumber: issueOrDiscussionNumber,
+      connectorId: connector.id,
+    },
+  });
+
+  if (!issueInDb) {
+    throw new Error("Issue not found in DB");
```

Comment:
Same here, would be nice to log the id that caused the failure so we can investigate.